tests/warn_conflict/tests/01-make.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+import subprocess
+import os
+
+
+def testfunc():
+    cross_gcc = "arm-none-eabi-gcc"
+
+    try:
+        devnull = open(os.devnull)
+        subprocess.Popen([cross_gcc],
+                         stdout=devnull, stderr=devnull).communicate()
+        output = subprocess.Popen(["make", "BOARD=stm32f4discovery"],
```

Comment:
I think it's safer to update `os.environ` for this.

Reply:
why ? this test is only meant to test for this particular platform (it doesn't use testrunner btw)

Reply:
I mean it just test compile time and not execution time.

Reply:
This has nothing to do with being about a particular platform, but how the `BOARD` macro is handed to `make`. I'm not sure, what happens if `BOARD` is already set in `env` when you define `BOARD` like it is here (I usually define it in the local `env` of the `make` process: `BOARD=stm32... make`)

Reply:
Indeed, I have different ouputs when `BOARD` is forced on the build command. I'll adapt.